private/itest.bzl

```diff
@@ -345,7 +355,7 @@ def _itest_service_group_impl(ctx):
     return [
         RunEnvironmentInfo(environment = _run_environment(ctx, service_specs_file)),
         DefaultInfo(runfiles = runfiles),
-        _ServiceGroupInfo(services = services),
+        _ServiceGroupInfo(services = services, deferred = False),
```

Comment:
i think `group` also needs a deferred prop, otherwise you can have (non-deferred service) depend on (group) depend on (deferred service) and it will defeat the check you added